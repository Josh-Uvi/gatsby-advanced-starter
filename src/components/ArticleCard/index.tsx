import React from "react";
import ReadButton from "./ArticleButton";
import ArticleInfo from "../ArticleInfo";
import { H3 } from "../../theme";
import { Post } from "../../types";

import {
  CardContainer,
  ArticleLink,
  Details,
  ArticleMeta,
  ArticleHeader,
  ArticleExcerpt,
  CardImage,
} from "./style";

type ArticleHeroCardProps = {
  post: Post;
  hero?: boolean;
};

const ArticleCard = ({ post, hero }: ArticleHeroCardProps): JSX.Element => (
  <CardContainer hero={hero}>
    <ArticleLink to={post.slug}>
      <CardImage src={post.coverImageUrl} alt={post.coverImageAlt} rounded />
    </ArticleLink>
    <Details hero={hero}>
      <ArticleMeta>
        <ArticleHeader>
          <ArticleInfo post={post} />
          <ArticleLink to={post.slug}>
            <H3>{post.title}</H3>
          </ArticleLink>
        </ArticleHeader>
        <ArticleLink to={post.slug}>
          <ArticleExcerpt hero={hero}>{post.excerpt}</ArticleExcerpt>
        </ArticleLink>
      </ArticleMeta>
      {hero && <ReadButton to={post.slug} />}
    </Details>
  </CardContainer>
);

export default ArticleCard;
