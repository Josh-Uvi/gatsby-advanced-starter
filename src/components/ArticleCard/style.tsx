import styled, { css } from "styled-components";
import { Link } from "gatsby";
import Image from "../shared/Image";

import { Body, styles } from "../../theme";

type CardStyleProps = { hero?: boolean };

export const ArticleLink = styled(Link)`
  text-decoration: none;
  color: inherit;

  width: 100%;
`;

export const CardImage = styled(Image)`
  width: 100%;
  height: 368px;

  @media (max-width: 500px) {
    height: 320px;
  }

  @media (max-width: 440px) {
    height: 270px;
  }
`;

export const CardContainer = styled.div`
  display: grid;
  grid-gap: 8px;

  ${({ hero }: CardStyleProps) =>
    hero &&
    css`
      @media (min-width: 1100px) {
        grid-column: span 2;
        grid-template-columns: 6fr 4fr;
        grid-auto-flow: column;
        grid-gap: 32px;
      }
    `}
`;

export const ArticleHeader = styled.div`
  display: grid;
  grid-gap: 12px;

  align-items: flex-start;
  align-content: start;
`;

export const ArticleExcerpt = styled(Body)`
  display: -webkit-box;
  -webkit-line-clamp: ${({ hero }: CardStyleProps) => (hero ? 5 : 2)};
  -webkit-box-orient: vertical;
  overflow: hidden;
`;

export const ArticleMeta = styled.div`
  display: grid;
  grid-gap: 12px;

  align-items: flex-start;
  align-content: start;
`;

export const Details = styled.div`
  display: grid;

  ${({ hero }: CardStyleProps) =>
    hero &&
    css`
      @media (min-width: 1100px) {
        align-content: space-between;
      }
    `}
`;

export const ReadButton = styled(Link)`
  text-decoration: none;
  color: var(--color-primary);

  ${styles.ButtonLabel}

  display: grid;
  grid-gap: 12px;
  grid-auto-flow: column;

  justify-content: start;
  align-items: center;

  @media (max-width: 1100px) {
    display: none;
  }
`;
