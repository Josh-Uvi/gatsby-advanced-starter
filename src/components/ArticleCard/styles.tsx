import styled, { css } from "styled-components";
import Image from "../shared/Image";

import { Body, constants } from "../../theme";

type CardStyleProps = { hero?: boolean };

export const Cover = styled(Image)`
  width: 100%;
  height: 368px;

  /* Tune the height on smaller screen resolutions */
  @media (max-width: 500px) {
    height: 320px;
  }

  @media (max-width: 440px) {
    height: 270px;
  }
`;

export const Wrapper = styled.div<CardStyleProps>`
  display: grid;
  grid-gap: 8px;

  ${({ hero }) =>
    hero &&
    css`
      @media (min-width: ${constants.breakpoints.lg}) {
        grid-column: span 2;
        grid-template-columns: 6fr 4fr;
        grid-gap: 32px;
      }
    `}
`;

export const Header = styled.div`
  display: grid;
  grid-gap: 12px;

  align-items: flex-start;
  align-content: start;
`;

export const Excerpt = styled(Body)<CardStyleProps>`
  display: -webkit-box;
  -webkit-line-clamp: ${({ hero }) => (hero ? 5 : 2)};
  -webkit-box-orient: vertical;
  overflow: hidden;
`;

export const Meta = styled.div`
  display: grid;
  grid-gap: 12px;

  align-items: flex-start;
  align-content: start;
`;

export const Details = styled.div<CardStyleProps>`
  display: grid;

  ${({ hero }) =>
    hero &&
    css`
      @media (min-width: var(--breakpoint-lg)) {
        align-content: space-between;
      }
    `}
`;
