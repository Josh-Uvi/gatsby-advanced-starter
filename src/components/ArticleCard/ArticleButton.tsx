import React from "react";
import { Exit } from "@styled-icons/boxicons-regular";

import { ReadButton } from "./style";

type ArticleButtonProps = { to: string };

const ArticleButton = ({ to }: ArticleButtonProps): JSX.Element => (
  <ReadButton to={to}>
    Read the article <Exit size={28} />
  </ReadButton>
);

export default ArticleButton;
