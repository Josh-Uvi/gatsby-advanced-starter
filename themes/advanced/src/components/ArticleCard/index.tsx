import React from "react";
import { getImage, IGatsbyImageData } from "gatsby-plugin-image";
import ReadButton from "./ArticleButton";
import ArticleInfo from "../ArticleInfo";
import { H3 } from "../../theme";
import { Post } from "../../types";
import { TransparentLink } from "../Links";

import ArticleCardSkeleton from "./Skeleton";

import * as S from "./styles";

type ArticleHeroCardProps = {
  post?: Post;
  hero?: boolean;
};

const ArticleCard = ({ post, hero }: ArticleHeroCardProps): JSX.Element => {
  // If no post is provided, render a skeleton view with a loading animation
  if (!post) return <ArticleCardSkeleton />;

  if (!post.coverImg)
    throw Error("Failed to render ArticleCard without Gatsby coverImg.");

  return (
    <S.Wrapper hero={hero}>
      <TransparentLink to={post.slug}>
        <S.Cover
          image={getImage(post.coverImg) as IGatsbyImageData}
          alt={post.coverImageAlt}
        />
      </TransparentLink>
      <S.Details hero={hero}>
        <S.Meta>
          <S.Header>
            <ArticleInfo post={post} />
            <TransparentLink to={post.slug}>
              <H3>{post.title}</H3>
            </TransparentLink>
          </S.Header>
          <TransparentLink to={post.slug}>
            <S.Excerpt hero={hero}>{post.excerpt}</S.Excerpt>
          </TransparentLink>
        </S.Meta>
        {hero && <ReadButton to={post.slug} />}
      </S.Details>
    </S.Wrapper>
  );
};

export default ArticleCard;
