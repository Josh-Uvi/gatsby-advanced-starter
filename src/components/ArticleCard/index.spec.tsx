import * as React from "react";
import "jest-styled-components";
import { render } from "@testing-library/react";

import { post } from "../../../test/sampleData";

import ArticleCard from "./index";

describe("component ArticleCard", () => {
  it("renders correctly", () => {
    expect.assertions(1);
    const { asFragment } = render(<ArticleCard post={post} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly as a hero card", () => {
    expect.assertions(1);
    const { asFragment } = render(<ArticleCard post={post} hero />);
    expect(asFragment()).toMatchSnapshot();
  });
});
