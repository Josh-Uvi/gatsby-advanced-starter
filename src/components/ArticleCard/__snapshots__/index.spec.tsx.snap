// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component ArticleCard renders correctly 1`] = `
<DocumentFragment>
  .c2 {
  object-fit: cover;
  box-shadow: inset 0px 0px 1px 1px rgba(0,0,0,0.1);
  border-radius: 10px;
}

.c15 {
  font-family: "Alfa Slab One",serif;
  font-style: normal;
  font-weight: normal;
  font-size: 26px;
  line-height: 32px;
  -webkit-letter-spacing: 0.15px;
  -moz-letter-spacing: 0.15px;
  -ms-letter-spacing: 0.15px;
  letter-spacing: 0.15px;
}

.c16 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 24px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
}

.c11 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 16px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  width: 100%;
}

.c3 {
  width: 100%;
  height: 368px;
}

.c0 {
  display: grid;
  grid-gap: 8px;
}

.c6 {
  display: grid;
  grid-gap: 12px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-content: start;
  -ms-flex-line-pack: start;
  align-content: start;
}

.c17 {
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.c5 {
  display: grid;
  grid-gap: 12px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-content: start;
  -ms-flex-line-pack: start;
  align-content: start;
}

.c4 {
  display: grid;
}

.c9 {
  color: var(--color-primary);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c14 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 10px;
  line-height: 16px;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  text-transform: uppercase;
  text-transform: uppercase;
}

.c14::before {
  content: "#";
}

.c10 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 16px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  text-transform: capitalize;
}

.c12 {
  color: var(--color-grey-700);
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c13 {
  display: grid;
  grid-auto-flow: column;
  grid-gap: 16px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (min-width:564px) {
  .c15 {
    font-family: "Alfa Slab One",serif;
    font-style: normal;
    font-weight: normal;
    font-size: 33px;
    line-height: 40px;
    -webkit-letter-spacing: 0.15px;
    -moz-letter-spacing: 0.15px;
    -ms-letter-spacing: 0.15px;
    letter-spacing: 0.15px;
  }
}

@media (min-width:564px) {
  .c16 {
    font-family: "Fira Sans",sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 18px;
    line-height: 28px;
    -webkit-letter-spacing: 0.5px;
    -moz-letter-spacing: 0.5px;
    -ms-letter-spacing: 0.5px;
    letter-spacing: 0.5px;
  }
}

@media (min-width:564px) {
  .c11 {
    font-family: "Fira Sans",sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 16px;
    -webkit-letter-spacing: 0.4px;
    -moz-letter-spacing: 0.4px;
    -ms-letter-spacing: 0.4px;
    letter-spacing: 0.4px;
  }
}

@media (max-width:500px) {
  .c3 {
    height: 320px;
  }
}

@media (max-width:440px) {
  .c3 {
    height: 270px;
  }
}

@media (min-width:564px) {
  .c14 {
    font-family: "Fira Sans",sans-serif;
    font-style: normal;
    font-weight: 500;
    font-size: 12px;
    line-height: 16px;
    -webkit-letter-spacing: 1.5px;
    -moz-letter-spacing: 1.5px;
    -ms-letter-spacing: 1.5px;
    letter-spacing: 1.5px;
    text-transform: uppercase;
  }
}

@media (min-width:564px) {
  .c10 {
    font-family: "Fira Sans",sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 16px;
    -webkit-letter-spacing: 0.4px;
    -moz-letter-spacing: 0.4px;
    -ms-letter-spacing: 0.4px;
    letter-spacing: 0.4px;
  }
}

<div
    class="c0"
  >
    <a
      class="c1"
      href="/big-sample-test"
    >
      <img
        alt="CoverImageAlt"
        class="c2 c3"
        src="5.jpg"
      />
    </a>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <div
              class="c8"
            >
              <a
                class="c9 c10"
                href="/categories/moar"
              >
                moar
              </a>
              <p
                class="c11 c12"
              >
                 ⋅ Mar 1, 2018 ⋅ 3 min read
              </p>
            </div>
            <div
              class="c13"
            >
              <a
                class="c9 c14"
                href="/tags/test"
              >
                test
              </a>
              <a
                class="c9 c14"
                href="/tags/huge"
              >
                huge
              </a>
            </div>
          </div>
          <a
            class="c1"
            href="/big-sample-test"
          >
            <h3
              class="c15"
            >
              Big Test
            </h3>
          </a>
        </div>
        <a
          class="c1"
          href="/big-sample-test"
        >
          <p
            class="c16 c17"
          >
            MDX Test Here's a color picker! NOTE: This "post" is based on  Markdown Cheatsheet  and is meant to test styling of Markdown generated…
          </p>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`component ArticleCard renders correctly as a hero card 1`] = `
<DocumentFragment>
  .c19 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  object-fit: cover;
  box-shadow: inset 0px 0px 1px 1px rgba(0,0,0,0.1);
  border-radius: 10px;
}

.c15 {
  font-family: "Alfa Slab One",serif;
  font-style: normal;
  font-weight: normal;
  font-size: 26px;
  line-height: 32px;
  -webkit-letter-spacing: 0.15px;
  -moz-letter-spacing: 0.15px;
  -ms-letter-spacing: 0.15px;
  letter-spacing: 0.15px;
}

.c16 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 24px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
}

.c11 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 16px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  width: 100%;
}

.c3 {
  width: 100%;
  height: 368px;
}

.c0 {
  display: grid;
  grid-gap: 8px;
}

.c6 {
  display: grid;
  grid-gap: 12px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-content: start;
  -ms-flex-line-pack: start;
  align-content: start;
}

.c17 {
  display: -webkit-box;
  -webkit-line-clamp: 5;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.c5 {
  display: grid;
  grid-gap: 12px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-content: start;
  -ms-flex-line-pack: start;
  align-content: start;
}

.c4 {
  display: grid;
}

.c18 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--color-primary);
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 18px;
  line-height: 16px;
  -webkit-letter-spacing: 1.25px;
  -moz-letter-spacing: 1.25px;
  -ms-letter-spacing: 1.25px;
  letter-spacing: 1.25px;
  text-transform: uppercase;
  display: grid;
  grid-gap: 12px;
  grid-auto-flow: column;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 {
  color: var(--color-primary);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c14 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 10px;
  line-height: 16px;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  text-transform: uppercase;
  text-transform: uppercase;
}

.c14::before {
  content: "#";
}

.c10 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 16px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  text-transform: capitalize;
}

.c12 {
  color: var(--color-grey-700);
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c13 {
  display: grid;
  grid-auto-flow: column;
  grid-gap: 16px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (min-width:564px) {
  .c15 {
    font-family: "Alfa Slab One",serif;
    font-style: normal;
    font-weight: normal;
    font-size: 33px;
    line-height: 40px;
    -webkit-letter-spacing: 0.15px;
    -moz-letter-spacing: 0.15px;
    -ms-letter-spacing: 0.15px;
    letter-spacing: 0.15px;
  }
}

@media (min-width:564px) {
  .c16 {
    font-family: "Fira Sans",sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 18px;
    line-height: 28px;
    -webkit-letter-spacing: 0.5px;
    -moz-letter-spacing: 0.5px;
    -ms-letter-spacing: 0.5px;
    letter-spacing: 0.5px;
  }
}

@media (min-width:564px) {
  .c11 {
    font-family: "Fira Sans",sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 16px;
    -webkit-letter-spacing: 0.4px;
    -moz-letter-spacing: 0.4px;
    -ms-letter-spacing: 0.4px;
    letter-spacing: 0.4px;
  }
}

@media (max-width:500px) {
  .c3 {
    height: 320px;
  }
}

@media (max-width:440px) {
  .c3 {
    height: 270px;
  }
}

@media (min-width:1100px) {
  .c0 {
    grid-column: span 2;
    grid-template-columns: 6fr 4fr;
    grid-auto-flow: column;
    grid-gap: 32px;
  }
}

@media (min-width:1100px) {
  .c4 {
    -webkit-align-content: space-between;
    -ms-flex-line-pack: space-between;
    align-content: space-between;
  }
}

@media (max-width:1100px) {
  .c18 {
    display: none;
  }
}

@media (min-width:564px) {
  .c14 {
    font-family: "Fira Sans",sans-serif;
    font-style: normal;
    font-weight: 500;
    font-size: 12px;
    line-height: 16px;
    -webkit-letter-spacing: 1.5px;
    -moz-letter-spacing: 1.5px;
    -ms-letter-spacing: 1.5px;
    letter-spacing: 1.5px;
    text-transform: uppercase;
  }
}

@media (min-width:564px) {
  .c10 {
    font-family: "Fira Sans",sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 16px;
    -webkit-letter-spacing: 0.4px;
    -moz-letter-spacing: 0.4px;
    -ms-letter-spacing: 0.4px;
    letter-spacing: 0.4px;
  }
}

<div
    class="c0"
  >
    <a
      class="c1"
      href="/big-sample-test"
    >
      <img
        alt="CoverImageAlt"
        class="c2 c3"
        src="5.jpg"
      />
    </a>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <div
              class="c8"
            >
              <a
                class="c9 c10"
                href="/categories/moar"
              >
                moar
              </a>
              <p
                class="c11 c12"
              >
                 ⋅ Mar 1, 2018 ⋅ 3 min read
              </p>
            </div>
            <div
              class="c13"
            >
              <a
                class="c9 c14"
                href="/tags/test"
              >
                test
              </a>
              <a
                class="c9 c14"
                href="/tags/huge"
              >
                huge
              </a>
            </div>
          </div>
          <a
            class="c1"
            href="/big-sample-test"
          >
            <h3
              class="c15"
            >
              Big Test
            </h3>
          </a>
        </div>
        <a
          class="c1"
          href="/big-sample-test"
        >
          <p
            class="c16 c17"
          >
            MDX Test Here's a color picker! NOTE: This "post" is based on  Markdown Cheatsheet  and is meant to test styling of Markdown generated…
          </p>
        </a>
      </div>
      <a
        class="c18"
        href="/big-sample-test"
      >
        Read the article 
        <svg
          aria-hidden="true"
          class="c19"
          fill="currentColor"
          focusable="false"
          height="28"
          viewBox="0 0 24 24"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.002 3h-14c-1.103 0-2 .897-2 2v4h2V5h14v14h-14v-4h-2v4c0 1.103.897 2 2 2h14c1.103 0 2-.897 2-2V5c0-1.103-.898-2-2-2z"
          />
          <path
            d="M11 16l5-4-5-4v3.001H3v2h8z"
          />
        </svg>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
